Waveform dump broken for iverilog

I am trying to get my verilog file to dump waves when compiled with iverilog. But when I run tests with the `WAVES=1` env variable I get the following error:
```
---------------------------------------------------------------------- Captured log call -----------------------------------------------------------------------
INFO     cocotb:simulator.py:291 Running command: iverilog -o /path/to/repo/cocotb-test-bug-demo/src/sim_build/dff.vvp -D COCOTB_SIM=1 -s dff -g2012 -s iverilog_dump /Users/stephan/environments/personal/pus/cocotb-test-bug-demo/src/dff.v
INFO     cocotb:simulator.py:302 error: Unable to find the root module "iverilog_dump" in the Verilog source.
INFO     cocotb:simulator.py:302      : Perhaps ``-s iverilog_dump'' is incorrect?
INFO     cocotb:simulator.py:302 2 error(s) during elaboration.
```

I think this got broken somewhere down the line after it got implemented because the original implementation in https://github.com/themperek/cocotb-test/pull/116 worked.

Demo repo available here: https://github.com/stephancill/cocotb-test-waves-bug-demo

--- a/cocotb_test/simulator.py
+++ b/cocotb_test/simulator.py
@@ -487,7 +487,9 @@ class Icarus(Simulator):
                     f.write("end\n")
                     f.write("endmodule\n")
 
-            verilog_sources += [dump_mod_file_name]
+            first_lib_src = next(iter(self.verilog_sources.items()))[1]
+            first_lib_src += [dump_mod_file_name]
+
             self.compile_args.extend(["-s", dump_mod_name])
             self.plus_args.append("-fst")
 

--- a/tests/test_dff.py
+++ b/tests/test_dff.py
@@ -8,7 +8,7 @@ tests_dir = os.path.dirname(__file__)
 @pytest.mark.skipif(os.getenv("SIM") == "ghdl", reason="Verilog not suported")
 # @pytest.mark.parametrize("seed", range(10))
 def test_dff_verilog():
-    run(verilog_sources=[os.path.join(tests_dir, "dff.sv")], toplevel="dff_test", module="dff_cocotb")  # sources  # top level HDL  # name of cocotb test module
+    run(verilog_sources=[os.path.join(tests_dir, "dff.sv")], toplevel="dff_test", module="dff_cocotb", waves=True)  # sources  # top level HDL  # name of cocotb test module
 
 
 @pytest.mark.skipif(os.getenv("SIM") == "verilator", reason="VHDL not suported")

